### PR TITLE
Issue #617: Remove -PublishXdocWithPush and -PublishXdoc command

### LINKED ISSUE #617: Remove -PublishXdocWithPush command
Child issue of https://github.com/checkstyle/checkstyle/issues/11782

-PublishXdocWithPush doesn't give the expected output. Mentors concluded to use a shell command to commit and push.

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Resolves #617

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
@romani @rnveach Please let me know what else needs to be removed?

**Review by rnveach — CHANGES_REQUESTED** [review]
Where are the removals in `XdocPublisher`?

Also it was my understanding from https://github.com/checkstyle/checkstyle/issues/11782#issuecomment-1173878150
> update script to do commit by shell commands

That `publishXdoc` would also be removed since it is the actual area that is not working by creating the commit.

**Comment by rnveach** [comment]
> releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java:[151,45] constructor XdocPublisher in class com.github.checkstyle.publishers.XdocPublisher cannot be applied to given types;

**Comment by rnveach** [comment]
Let me know when this PR is ready to review.

**Comment by Rahulkhinchi03** [comment]
@rnveach PR is ready from my side. Is there anything left to remove?

File: releasenotes-builder/src/main/java/com/github/checkstyle/publishers/XdocPublisher.java [hunk 1]
```diff
@@ -57,56 +45,16 @@ public class XdocPublisher {
...
-    }
-
-    /**
-     * Publish release notes.
-     *
-     * @throws IOException if problem with access to files appears.
-     * @throws GitAPIException for problems with jgit.
-     */
-    public void publish() throws IOException, GitAPIException {
-        changeLocalRepoXdoc();
```
Comment (rnveach) [inline releasenotes-builder/src/main/java/com/github/checkstyle/publishers/XdocPublisher.java:92]: private method `changeLocalRepoXdoc` remains but is not being used anywhere.

Comment (Rahulkhinchi03) [reply releasenotes-builder/src/main/java/com/github/checkstyle/publishers/XdocPublisher.java:92]: Right, Did Changes.

**Review by rnveach — APPROVED** [review]

**Pull request merged**